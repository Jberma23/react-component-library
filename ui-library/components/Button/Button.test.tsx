import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from ".";

// Since this is just a sample component, the tests below are not intended to be comprehensive
describe("Button", () => {
  it("renders with defaults", () => {
    const { asFragment } = render(<Button>test button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("appends additional classnames", () => {
    render(<Button className="my-test-class">test button</Button>);
    const buttonClass = screen.getByRole("button")?.getAttribute("class");
    expect(buttonClass).toContain("my-test-class");
  });

  it("can be disabled", () => {
    render(<Button disabled={true}>test button</Button>);
    const buttonEl = screen.getByRole("button");
    const buttonClass = buttonEl?.getAttribute("class");
    expect(buttonEl).toHaveProperty("disabled", true);
    expect(buttonClass).toContain("opacity-40");
  });

  it("passes down additional button props", async () => {
    const mockClickHandler = jest.fn();
    render(<Button onClick={mockClickHandler}>test button</Button>);
    const buttonEl = screen.getByRole("button");
    await userEvent.click(buttonEl);
    expect(mockClickHandler).toHaveBeenCalled();
  });
});
