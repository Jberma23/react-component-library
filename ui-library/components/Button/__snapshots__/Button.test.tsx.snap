// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with defaults 1`] = `
<DocumentFragment>
  <button
    class="border-1 border-transparent rounded-3 ring-offset-1 focus-visible:ring-2 focus-visible:ring-brand transition-colors duration-default bg-gray-200 hover:bg-gray-300 active:bg-gray-400 text-black h-[2.5rem] px-[1rem] py-0 leading-body font-semibold text-body-regular"
  >
    <div
      class="flex flex-row gap-xxs"
    >
      <div
        class="flex-1"
      >
        test button
      </div>
    </div>
  </button>
</DocumentFragment>
`;
